src/test/java/com/example/demo/OrderServiceTests.java

```diff
@@ -1,27 +1,238 @@
 package com.example.demo;
 
 import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.OrderbookItem;
 import com.example.demo.matcher.services.OrderService;
 import com.example.demo.matcher.models.Order;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockito.Spy;
 import org.springframework.boot.test.context.SpringBootTest;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.spy;
 
 @SpringBootTest
 public class OrderServiceTests {
+    @Spy
     OrderService orderService;
+
     @BeforeEach
     void InitialiseOrderService() {
-        orderService = new OrderService();
+        orderService = spy(new OrderService());
```

Comment:
I see what you're doing now. I was looking at this: https://www.baeldung.com/mockito-annotations when I reviewed this.

You could try getting rid of the `@Spy` annotation on line 23.

Then do something like this in the before each:
```java
@Spy
orderService = new OrderService();
```

I'm not sure if that works. Try it, if not we can leave it, again it doesn't really make a difference :)

Reply:
![image](https://user-images.githubusercontent.com/118171430/211025458-03f2ffc3-92db-4db6-ad82-1cd4d9541d69.png)

Reply:
I'll get rid of the annotation for now. I had a little google and didn't find any quick answers